client/browser/src/shared/repo/backend.tsx

```diff
@@ -103,7 +103,10 @@ export function retryWhenCloneInProgressError<T>(): (v: Observable<T>) => Observ
         )
 }
 
-const trimRepoName = ({ repoName, ...rest }) => ({ ...rest, repoName: repoName.replace(/.git$/, '') })
+const trimRepoName = <T extends { repoName: string }>({ repoName, ...rest }: T) => ({
```

Comment:
```suggestion
const trimRepoName = <T extends { repoName: string }>({ repoName, ...rest }: T): T => ({
```

It's always a good idea to add return types to functions (that are not inline) to verify the return type is what is intended. It also often makes displayed type names clearer than what TS would infer.